src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -140,20 +96,21 @@ const TopicFragment = ({
     }
   };
 
-  const handleFollow = async (from, willFollow, item, index) => {
-    handleTopic(from, willFollow, item, index);
+  const handleFollow = async (from, willFollow, item) => {
+    handleTopic(from, willFollow, item);
 
     try {
       await followTopic(item?.name);
     } catch (e) {
-      handleTopic(from, !willFollow, item, index);
+      handleTopic(from, !willFollow, item);
     }
     if (searchText.length > 0) fetchData();
   };
 
   const __handleOnTopicPress = (item) => {
     const navigationParam = {
-      id: convertTopicNameToTopicPageScreenParam(item.name)
+      id: convertTopicNameToTopicPageScreenParam(item.name),
+      isFollowing: item.following
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here seem to be an improvement in terms of code readability and maintainability. The removal of the `index` parameter from the `handleFollow` function simplifies its usage. However, it's important to ensure that the `followTopic` function handles all edge cases and performs as expected.



<!-- This is an auto-generated comment by OSS CodeRabbit -->